apps/hash-graph/lib/graph/src/store/query/filter.rs

```diff
@@ -659,7 +1011,14 @@ mod tests {
           ]
         }};
 
-        test_filter_representation(&Filter::for_right_entity_by_entity_id(entity_id), &expected);
+        test_filter_representation(
+            &Filter::for_knowledge_graph_edge_by_entity_id(
+                entity_id,
+                KnowledgeGraphEdgeKind::HasRightEntity,
+                true,
+            ),
```

Comment:
Just want to say that this seems to be more accurate to model the above query. In the old code `Filter::for_right_entity_by_entity_id` didn't really capture whether or not this was a reversed link or not. This new filter syntax is much more clear on that :+1: